### PR TITLE
Use NWPathMonitor to get interfaces in order of preference for socket protector on apple platforms

### PR DESCRIPTION
**Pull request by tomaszklak**
Using only interfaces with the router property is incorrect since when using static IPs it's possible to have an interface without router property. Additionally, interfaces returned by the dynamic store are not in any particular order. After this, change, we use the interfaces collected from the NWPathMonitor. Those are ordered in the same way as the OS would prefer them.

I will try to add some sort of tests when merging to main.

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] changelog.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: crates/telio-sockets/Cargo.toml [hunk 1]
```diff
@@ -31,6 +31,9 @@ version-compare = "0.1"
 debug_panic = "0.2.1"
 nix = "0.26.2"
 system-configuration = { git = 'https://github.com/NordSecurity/system-configuration-rs.git' }
+network-framework-sys = "0.1"
```
[the remark this answers is not in the document]
Comment (tomaszklak) [reply crates/telio-sockets/Cargo.toml:34]: I think it already is in that section, or am I missing something?

[the remark this answers is not in the document]
Comment (matislovas) [reply crates/telio-sockets/Cargo.toml:34]: My bad! Forgot to expand the file :smile:

**Comment by matislovas** [comment]
+0.5

**Review by dfetti — COMMENTED** [review]
+0.5 Looks good, but I can't give you more than that because I'm not familiar at all with mac/ios APIs.

**Review by Jauler — APPROVED** [review]
+1.0

**Pull request merged**